src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -1811,22 +1811,12 @@ private void DecorateItemsWithMetadataFromProjectItemElement(ProjectItemElement
                     include.Add(itemElement.Include);
                     ItemsAndMetadataPair itemsAndMetadataFromInclude = ExpressionShredder.GetReferencedItemNamesAndMetadata(include);
 
-                    if (itemsAndMetadataFromInclude.Items != null && itemsAndMetadataFromInclude.Items.Count > 0)
-                    {
-                        needToProcessItemsIndividually = true;
-                    }
-                    else
-                    {
-                        // If there is bare built-in metadata, we must always run items individually, as that almost
-                        // always differs between items.
-
-                        // UNDONE: When batching is implemented for real, we need to make sure that
-                        // item definition metadata is included in all metadata operations during evaluation
-                        if (itemsAndMetadataFound.Metadata.Values.Count > 0)
-                        {
-                            needToProcessItemsIndividually = true;
-                        }
-                    }
+                    // If there is bare built-in metadata, we must always run items individually, as that almost
```

Comment:
IIUC, the earlier comment on lines 1808-1809 and this can be "merged" to be something like:

```
// If there is bare metadata of any kind, we must always run items individually. The two cases are:
//
// 1. If the Include involved an item list, we need to run items individually, as even non-built-in
//     metadata might differ between items
// 2. For bare *built-in* metadata, we need to run items individually, as that almost always
//     differs between items
//
// UNDONE: When batching is implemented for real, we need to make sure that
// item definition metadata is included in all metadata operations during evaluation
```

And the same can be done for the changes in `LazyItemOperation.cs` too.

Somebody familiar with this code will need to ensure that the above is correct though!